### DISCUSSION TITLE
Can we change the text for upload content?

### DISCUSSION THREAD

**Discussion by pranavirabatti** [Sunbird-Knowlg/Community#33] [General Discussions]
@pallakartheekreddy Any way to modify this text? 

![image](https://user-images.githubusercontent.com/22250006/178428459-361abcfc-bf41-49a8-8626-8e479a9ce427.png)

**Comment by vaibhavbhuva** [comment]
@pranavirabatti What you shown in the screeshot is not configurable. It's hard-coded at plugin level(used by editor).
Please check the below link for your refereance :
https://github.com/project-sunbird/sunbird-content-plugins/blob/23c335a90772cb308ddcafcd10651c1e08fbb21f/org.ekstep.uploadcontent-1.5/editor/upload.html#L102

cc: @vinukumar-vs @pallakartheekreddy

**Reply by coolbung** [reply]
@pranavirabatti -- please make a PR to the plugin code and portal by substituting the hardcoded DIKHSA with the name of the tenant. 

@vaibhavbhuva -- please help with the steps to test / fix locally and also deploy this fix

**Reply by pramodkvarma** [reply]
@coolbung 
@vinukumar-vs @SanthoshVasabhaktula how are these translated when UI language is changed? Aren't these texts in a bundle already?

**Reply by SanthoshVasabhaktula** [reply]
@pramodkvarma - Texts should be in bundle and configurable when the editor/player is instantiated by design. I presume this would have fallen through the cracks when we were building a lot of plugins for player & editor functionality with time constraints.

Translation when the UI language is changed has not been supported when the initial version of player and editors are built due to the following reasons:
1. Editor and player plugins were built as pure JS or limited dependency on Angular 1
2. They are integrated into portal via IFrame as the tech stack (Angular versions) is not compatible. The language preference is therefore not automatically picked up

When we have done the architecture upgrade to the players, we have handled the above limitations and now the players (except ECML) support resource bundles and are integrated natively to the portal. We haven't been able to pick up the similar architecture upgrade to the editors due to bandwidth constraints and other priorities being critical.

@vinukumar-vs @rhwarrier - Let us add these to the issue tracker and prioritize it for next quarter. We should also seek community support in upgrading the editors.

**Reply by pramodkvarma** [reply]
Thanks @SanthoshVasabhaktula for clarifying. Since @coolbung has offered to a PR, maybe we can get it fixed faster. 

@coolbung and @pranavirabatti please move the string to the bundle like all other labels while doing the PR. 

@vinukumar-vs please help @pranavirabatti identify the bundle to which it should go and then help with merging PR.
